tests/python/unittest/test_random.py

```diff
@@ -552,6 +554,56 @@ def compute_expected_prob():
     mx.test_utils.assert_almost_equal(exp_cnt_sampled.asnumpy(), exp_cnt[sampled_classes].asnumpy(), rtol=1e-1, atol=1e-2)
     mx.test_utils.assert_almost_equal(exp_cnt_true.asnumpy(), exp_cnt[true_classes].asnumpy(), rtol=1e-1, atol=1e-2)
 
+@with_seed()
+def test_shuffle():
+    def hash(arr):
```

Comment:
Concerned with the speed of hash function.

Reply:
We need to distinguish the order of the elements in the array and it seems not a bottleneck.

Reply:
I understand your logic here. I think there are too many blocking calls involved (`asscalar`). We want to do the best to squeeze the time of unit tests. Since your input array is small, you can use `hash(str(arr))` which should be more efficient (in terms of speed) and reliable (in terms of hash value collision) than the current implementation.